File: array/1583_count_unhappy_friends/1583_count_unhappy_friends.go
package main 


func unhappyFriends(n int, preferences [][]int, pairs [][]int) int {
	dd := make(map[int][]int)

	for _, pair := range pairs {
		i, x := pair[0], pair[1]
		for idx, j := range preferences[i] {
			if j == x {
				dd[i] = preferences[i][:idx]
			}
		}
		for idx, j := range preferences[x] {
			if j == i {
				dd[x] = preferences[x][:idx]
			}
		}
	}

	ans := 0
    fmt.Println(dd)

	for i := range dd {
        out:
		for _, x := range dd[i] {
			for _, j := range dd[x] {
				if i == j {
					ans++
					break out
				}
			}
		}
	}
	return ans
}

File: array/1583_count_unhappy_friends/1583_count_unhappy_friends.py
from typing import List
class Solution:
    def unhappyFriends(self, n: int, preferences: List[List[int]], pairs: List[List[int]]) -> int:
        dd = {}
        
        for i,x in pairs:
            dd[i] = preferences[i][:preferences[i].index(x)]
            dd[x] = preferences[x][:preferences[x].index(i)]
        
        ans = 0
            
        for i in dd:
            for x in dd[i]:
                if i in dd[x]:
                    ans += 1
                    break
        
        return ans


n = 4
preferences = [[1,2,3],[3,2,0],[3,1,0],[1,2,0]]
pairs = [[0,1],[2,3]]
res = Solution().unhappyFriends(n, preferences, pairs)
print(res)